### ISSUE TITLE
Modernize test framework

### ISSUE THREAD

**Issue by mhamann** [indexzero/nconf#282]
Convert existing `vows.js` based test suite to something a bit more common and well-known. This will help newer contributors understand how to write an effective test, which will lower the barrier to entry.

Potential frameworks include: ava, tap, and mocha.

**Comment by AdrieanKhisbe** [comment]
after spending some time with the test I think it's definitively a must have.
Besides vows is unmaintained for years.

For frameworks I had in mind **ava**, **jest**, and **mocha**.
The two first are the most trending theses days, and probably more jest than ava with react traction.

I would rather tend to jest because it has a *describe*/*it* nested test structure that ava does not have.

(https://stackshare.io/stackups/jest-vs-ava)

**Comment by AdrieanKhisbe** [comment]
@mhamann Shall I start to translate some of the test (some in ava, some in jest to serve as base of comparison?)

**Comment by mhamann** [comment]
Yep, go for it. If Jest feels like a good fit, I am fine moving forward with that. I have no real attachment to either.

**Comment by AdrieanKhisbe** [comment]
I start working on jest migration, you can see the first result there: 
https://github.com/indexzero/nconf/compare/master...AdrieanKhisbe:migrating-test-to-jest?expand=1

Waiting for your go @mhamann to go further. :)

**Comment by mhamann** [comment]
LGTM.

**Comment by AdrieanKhisbe** [comment]
Voilà I made the whole transition, tell me what you think @mhamann 😃 

(by the way, did you add chance to advance on the 0.10? =) )

**Comment by AdrieanKhisbe** [comment]
@mhamann up :)

**Comment by mhamann** [comment]
Thanks--PR looking good. Hoping to finish reviewing by end of week. I'll also get 0.10 then as well. Sorry for the delay--life has been a bit crazy the past couple of weeks. Appreciate the reminder!

**Comment by AdrieanKhisbe** [comment]
No problem, take the time you need. 
and courage!! :)

**Comment by AdrieanKhisbe** [comment]
Hey @mhamann !
I hope the crazy times are fading away.
Tell me if you need help on thé nconf part =)
